server/controllers/addUser.js

```diff
@@ -2,20 +2,25 @@ const { hashPassword, signupSchema } = require('../utils/validations');
 const { addUserQuery, getUserId } = require('../database/queries');
 
 const addUser = (req, res, next) => {
-  const { error, value: { username, email, password } } = signupSchema.validate(req.body);
+  const {
+    error, value: {
+      username, email, password, userRole,
+    },
+  } = signupSchema.validate(req.body);
 
   if (error) {
     res
       .status(400)
       .json({ msg: error.details[0].message });
   } else {
     hashPassword(password)
-      .then((hashed) => addUserQuery(username, email, hashed)
+      .then((hashed) => addUserQuery(username, email, hashed, userRole)
```

Comment:
Not sure why you did those changes in addProduct branch, 
Even though did you think what might happen if you try to add a new user with an existing email?!